libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
```

Comment:
too many counters, you can achieve the same with just one counter and increment the pointers to avoid ptr arithmetics on each iteration. modern compilers should be able to optimize this, but its always good to write optimal code that way